src/types/type_row.rs

```diff
@@ -32,7 +31,7 @@ impl<T: Display + TypeRowElem> Display for TypeRow<T> {
 
 // TODO some of these, but not all, will probably want exposing via
 // pyo3 wrappers eventually.
-impl<T: TypeRowElem> TypeRow<T> {
+impl TypeRow {
```

Comment:
P-low issue: replace these with a `delegate!` call (so we don't miss the #inline directives :) )